addOns/kotlin/src/main/java/org/zaproxy/zap/extension/kotlin/KotlinEngineWrapper.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2020 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.kotlin;
+
+import java.util.Collections;
+import java.util.List;
+import javax.script.ScriptContext;
+import javax.script.ScriptEngine;
+import javax.script.ScriptException;
+import javax.swing.ImageIcon;
+import kotlin.script.experimental.jsr223.KotlinJsr223DefaultScriptEngineFactory;
+import kotlin.script.experimental.jsr223.KotlinJsr223DefaultScriptEngineFactoryKt;
+import org.fife.ui.rsyntaxtextarea.SyntaxConstants;
+import org.zaproxy.zap.control.ExtensionFactory;
+import org.zaproxy.zap.extension.script.DefaultEngineWrapper;
+
+public class KotlinEngineWrapper extends DefaultEngineWrapper {
+
+    static {
+        System.setProperty(
+                KotlinJsr223DefaultScriptEngineFactoryKt
+                        .KOTLIN_JSR223_RESOLVE_FROM_CLASSLOADER_PROPERTY,
+                "true");
+    }
+
+    private ClassLoader classLoader;
+
+    public KotlinEngineWrapper() {
+        this(null);
+    }
+
+    public KotlinEngineWrapper(ClassLoader engineClassLoader) {
```

Comment:
- `engineClassLoader` → `fallbackClassLoader`, should replace `ExtensionFactory.getAddOnLoader()` (which can be passed in the default constructor).
 - Should be package accessible, just for testing?

Reply:
Yeah I was trying to make this configurable for testing as `ExtensionFactory.getAddOnLoader()` was throwing an NPE but perhaps I just need to setup my test better?

Reply:
No, it makes sense to allow to pass a custom class loader (really no need to mock `ExtensionFactory`).